invenio_rest/__init__.py

```diff
@@ -95,6 +95,79 @@
 
 Now you can launch your server and request it on the `/records` endpoint, as
 described in :any:`examplesapp`.
+
+Building REST APIs for Invenio
+------------------------------
+Following is a quick overview over which tools we are currently using for
+building REST APIs. Before implementing your REST API, do take a look at some
+of the existing REST APIs already implemented in Invenio to get some
+inspiration.
+
+In Invenio we have decided not to use some of the existing Flask extensions for
+building REST APIs since mostly these extensions are not very flexible and
+there are many existing Python libraries that does a much better job at
+the individual tasks.
+
+Flask application
+~~~~~~~~~~~~~~~~~
+Invenio's REST API is running in its own Flask application. This ensures that
+the REST API can be running on machines independently of the UI application
+and also ensures that e.g. error handling is much simpler compared to having a
+mixed REST API/UI application.
+
+Views
+~~~~~
+Views for REST APIs are built using standard Flask blueprints. We use
+:class:`~flask.views.MethodView` for HTTP method based dispatching, and in
+particular we use Invenio-REST's subclass
+:class:`~.views.ContentNegotiatedMethodView` which takes care of selecting the
+right serializer based on HTTP content negotiation.
+
+Versioning
+~~~~~~~~~~
+Versioning of the REST API is primarily achieved through HTTP content
+negotiation. I.e. you define a new MIME type that your clients explicitly
+request (using :class:`~.views.ContentNegotiatedMethodView`).
+
+Serialization/Deserialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For serialization/deserialization we primarily use
+`Marshmallow <http://marshmallow.readthedocs.io/en/latest/>`_ which takes care
+of ensuring that e.g. timestamps are ISO8601 formatted in UTC when serializing
+to JSON, and correctly deserialize timestamps into Python datetime objects.
+
+`Invenio-Records-REST <http://invenio-records-rest.readthedocs.io/en/latest/>`_
```

Comment:
It may be better to remove `/en/latest` here and elsewhere in the PR to keep only `http://invenio-records-rest.readthedocs.io/` as the visible canonical URL.